generate-stackbrew-pr.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+set -e
+. functions.sh
+
+GITHUB_USERNAME="nodejs-docker-bot"
+gitpath="../docker-images"
+IMAGES_FILE="library/node"
+REPO_NAME="official-images"
+ORIGIN_SLUG="$GITHUB_USERNAME/$REPO_NAME"
+UPSTREAM_SLUG="docker-library/$REPO_NAME"
+
+function updated() {
+	local versions
+	local images_changed
+
+	IFS=' ' read -ra versions <<< "$(IFS=','; get_versions)"
+	images_changed=$(git show --name-only "$TRAVIS_COMMIT" "${versions[@]}")
+
+	if [ -z "$images_changed" ]; then
+		return 1
+	else
+		return 0
+	fi
+}
+
+function permission_check() {
+	auth="$(curl -H "Authorization: token $GITHUB_API_TOKEN" \
+		-s \
+		"https://api.github.com")"
+	if [ "$(echo "$auth" | jq .message)" = "\"Bad credentials\"" ]; then
+		fatal "Authentication Failed! Invalid \$GITHUB_API_TOKEN"
+	fi
+
+	auth="$(curl -H "Authorization: token $GITHUB_API_TOKEN" \
+		-s \
+		"https://api.github.com/repos/$ORIGIN_SLUG/collaborators/$GITHUB_USERNAME/permission")"
+	if [ "$(echo "$auth" | jq .message)" != "null" ]; then
+		fatal "\$GITHUB_API_TOKEN can't push to https://github.com/$ORIGIN_SLUG.git"
+	fi
+}
+
+function setup_git_author() {
+	GIT_AUTHOR_NAME="$(git show -s --format="%aN" "$TRAVIS_COMMIT")"
+	GIT_AUTHOR_EMAIL="$(git show -s --format="%aE" "$TRAVIS_COMMIT")"
+	GIT_COMMITTER_NAME="$(git show -s --format="%cN" "$TRAVIS_COMMIT")"
+	GIT_COMMITTER_EMAIL="$(git show -s --format="%cN" "$TRAVIS_COMMIT")"
+
+	export GIT_AUTHOR_NAME GIT_AUTHOR_EMAIL GIT_COMMITTER_NAME GIT_COMMITTER_EMAIL
+}
+
+function message() {
+	echo "Node: $TRAVIS_COMMIT_MESSAGE"
+}
+
+function pr_payload() {
+	local escaped_message
+	IFS=' ' read -ra escaped_message <<< "$TRAVIS_COMMIT_MESSAGE"
+	escaped_message="$(printf '%q' "${escaped_message[@]}")"
+	echo '{
+		"title": "Node: '"$escaped_message"'",
+		"body": "Commit: nodejs/docker-node@'"$TRAVIS_COMMIT"'",
+		"head": "'"$GITHUB_USERNAME"':master",
+		"base": "master"
+	}'
+}
+
+if updated; then
+
+	permission_check
+
+	# Set Git User Info
+	setup_git_author
+
+	info "Cloning..."
+	git clone https://github.com/docker-library/official-images.git $gitpath 2> /dev/null
```

Comment:
I think we should use shallow clone here, it won't be unable to push back, if you can't push back, the reason why would be the gap of missing commits between the target and source.

There are 8k+ commits in [docker-library/official-images/](https://github.com/docker-library/official-images/), most of them are those we don't need, set the depth to 100 would be helpful here, to save the waste every single round.

Reply:
I was under the impression that shallow clones can't push back but that changed since git v1.9. I'll try it out!